com.unity.netcode.adapter.utp/Tests/Runtime/TransportTests.cs

```diff
@@ -305,6 +305,33 @@ public IEnumerator ReceiveMultipleClients([ValueSource("k_DeliveryParameters")]
 
             yield return null;
         }
+
+        // Check that simulator parameters are effective. We only check with the drop rate, because
+        // that's easy to check and we only really want to make sure the simulator parameters are
+        // configured properly (the simulator pipeline stage is already well-tested in UTP).
+        [UnityTest]
+        [UnityPlatform(include = new[] { RuntimePlatform.OSXEditor, RuntimePlatform.WindowsEditor, RuntimePlatform.LinuxEditor })]
+        public IEnumerator SimulatorParametersAreEffective()
+        {
+            InitializeTransport(out m_Server, out m_ServerEvents);
+            InitializeTransport(out m_Client1, out m_Client1Events);
+
+            m_Server.SetDebugSimulatorParameters(0, 0, 100);
+
+            m_Server.StartServer();
+            m_Client1.StartClient();
+
+            yield return WaitForNetworkEvent(NetworkEvent.Connect, m_Client1Events);
+
+            var data = new ArraySegment<byte>(new byte[] { 42 });
+            m_Client1.Send(m_Client1.ServerClientId, data, NetworkDelivery.Reliable);
+
+            yield return new WaitForSeconds(MaxNetworkEventWaitTime);
+
+            Assert.AreEqual(1, m_ServerEvents.Count);
```

Comment:
My understanding is that you are sending a packet but it should be dropped (I am assuming on the `server` pipeline end) , will you still receive the event ? If so, wouldn't it be the case regardless of the simulator config ?

Reply:
If no packet is received, then there wouldn't be any event to propagate. The adapter is not aware of packets being dropped by the simulator pipeline. That line here is a bit confusing because there's still the `Connect` event in the server events list (a test checking if the event was received would test that the count is 2).